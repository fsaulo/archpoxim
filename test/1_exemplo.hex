0xDC000007
0x00000000
0x00000000
0x00000000
0x00000000
0x00000000
0x00000000
0x00000000
0x00323456
0x04500000
0x08611000
0x1002130A
0x0C821800
0x10A41800
0x10C52900
0x10E62A00
0x11073B01
0x11283C00
0x11494D02
0x1169460A
0x118A5703
0x141CE800
0x19A12800
0x1DC23000
0x21E70000
0x26104000
0x4A310001
0x4E52FFFF
0x52710002
0x56930002
0x5AB30003
0x5C150020
0x62C00023
0x66E00011
0x6B000008
0x6EC00023
0x72E00011
0x77000008
0xA8000000
0xAC000000
0xB0000000
0xB4000000
0xB8000000
0xBC000000
0xC0000000
0xC4000000
0xC8000000
0xCC000000
0xD0000000
0xD4000000
0xD8000000
0xDC000000
0xE0000000
0xFC000000
