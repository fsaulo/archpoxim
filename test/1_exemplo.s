.text
	init:
		bun main
		.align 5
	main:
		mov r1, 0x123456
		movs r2, -1048576
		add r3, r1, r2
		sla r0, r2, r2, 11
		sub r4, r2, r3
		mul r5, r4, r3
		sll r6, r5, r5, 1
		muls r7, r6, r5
		sla r8, r7, r7, 2
		div r9, r8, r7
		srl r10, r9, r9, 3
		divs r10, r11, r9, r8
		sra r12, r10, r10, 4
		cmp ir, pc
		and r13, r1, r5
		or r14, r2, r6
		not r15, r7
		xor r16, r16, r8
		addi r17, r17, +1
		subi r18, r18, -1
		muli r19, r17, 2
		divi r20, r19, 2
		modi r21, r19, 3
		cmpi r21, 0x20
		l8 r22, [main + 3]
		l16 r23, [main + 1]
		l32 r24, [main]
		s8 [main + 3], r22
		s16 [main + 1], r23
		s32 [main], r24
		bae 0
		bat 0
		bbe 0
		bbt 0
		beq 0
		bge 0
		bgt 0
		biv 0
		ble 0
		blt 0
		bne 0
		bni 0
		bnz 0
		bun 0
		bzd 0
		int 0
