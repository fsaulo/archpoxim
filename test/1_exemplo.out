[START OF SIMULATION]
0x00000000:	bun 7                    	PC=0x00000020
0x00000020:	mov r1,1193046           	R1=0x00123456
0x00000024:	movs r2,-1048576         	R2=0xFFF00000
0x00000028:	add r3,r1,r2             	R3=R1+R2=0x00023456,SR=0x00000001
0x0000002C:	sla r0,r2,r2,10          	R0:R2=R0:R2<<11=0x0000000080000000,SR=0x00000001
0x00000030:	sub r4,r2,r3             	R4=R2-R3=0x7FFDCBAA,SR=0x00000008
0x00000034:	mul r0,r5,r4,r3          	R0:R5=R4*R3=0x0000000023F4F31C,SR=0x00000008
0x00000038:	sll r6,r5,r5,0           	R6:R5=R6:R5<<1=0x0000000047E9E638,SR=0x00000008
0x0000003C:	muls r0,r7,r6,r5         	R0:R7=R6*R5=0x0000000000000000,SR=0x00000040
0x00000040:	sla r8,r7,r7,1           	R8:R7=R8:R7<<2=0x0000000000000000,SR=0x00000040
0x00000044:	div r0,r9,r8,r7          	R0=R8%R7=0x00000000,R9=R8/R7=0x00000000,SR=0x00000060
0x00000048:	srl r10,r9,r9,2          	R10:R9=R10:R9>>3=0x0000000000000000,SR=0x00000060
0x0000004C:	divs r10,r11,r9,r8       	R10=R9%R8=0x00000000,R11=R9/R8=0x00000000,SR=0x00000060
0x00000050:	sra r12,r10,r10,3        	R12:R10=R12:R10>>4=0x0000000000000000,SR=0x00000060
0x00000054:	cmp ir,pc                	SR=0x00000020
0x00000058:	and r13,r1,r5            	R13=R1&R5=0x00002410,SR=0x00000020
0x0000005C:	or r14,r2,r6             	R14=R2|R6=0x80000000,SR=0x00000030
0x00000060:	not r15,r7               	R15=~R7=0xFFFFFFFF,SR=0x00000030
0x00000064:	xor r16,r16,r8           	R16=R16^R8=0x00000000,SR=0x00000060
0x00000068:	addi r17,r17,1           	R17=R17+0x00000001=0x00000001,SR=0x00000020
0x0000006C:	subi r18,r18,-1          	R18=R18-0xFFFFFFFF=0x00000001,SR=0x00000020
0x00000070:	muli r19,r17,2           	R19=R17*0x00000002=0x00000002,SR=0x00000020
0x00000074:	divi r20,r19,2           	R20=R19/0x00000002=0x00000001,SR=0x00000000
0x00000078:	modi r21,r19,3           	R21=R19%0x00000003=0x00000002,SR=0x00000000
0x0000007C:	cmpi r21,32              	SR=0x00000011
0x00000080:	l8 r22,[r0+35]           	R22=MEM[0x00000023]=0x56
0x00000084:	l16 r23,[r0+17]          	R23=MEM[0x00000022]=0x3456
0x00000088:	l32 r24,[r0+8]           	R24=MEM[0x00000020]=0x00323456
0x0000008C:	s8 [r0+35],r22           	MEM[0x00000023]=R22=0x56
0x00000090:	s16 [r0+17],r23          	MEM[0x00000022]=R23=0x3456
0x00000094:	s32 [r0+8],r24           	MEM[0x00000020]=R24=0x00323456
0x00000098:	bae 0                    	PC=0x0000009C
0x0000009C:	bat 0                    	PC=0x000000A0
0x000000A0:	bbe 0                    	PC=0x000000A4
0x000000A4:	bbt 0                    	PC=0x000000A8
0x000000A8:	beq 0                    	PC=0x000000AC
0x000000AC:	bge 0                    	PC=0x000000B0
0x000000B0:	bgt 0                    	PC=0x000000B4
0x000000B4:	biv 0                    	PC=0x000000B8
0x000000B8:	ble 0                    	PC=0x000000BC
0x000000BC:	blt 0                    	PC=0x000000C0
0x000000C0:	bne 0                    	PC=0x000000C4
0x000000C4:	bni 0                    	PC=0x000000C8
0x000000C8:	bnz 0                    	PC=0x000000CC
0x000000CC:	bun 0                    	PC=0x000000D0
0x000000D0:	bzd 0                    	PC=0x000000D4
0x000000D4:	int 0                    	CR=0x00000000,PC=0x00000000
[END OF SIMULATION]
